### PR TITLE
Allow Explorer.PolarsBackend.Series.pow/2 on (number, series)

### LINKED ISSUE #389: Allow `Explorer.PolarsBackend.Series.pow/2` on (number, series)
```elixir
test "pow of a series with an integer scalar value on the left-hand side" do
  s1 = Series.from_list([1, 2, 3])

  s2 = Series.pow(2, s1)

  assert s2.dtype == :integer
  assert Series.to_list(s2) == [2, 4, 8]
end

test "pow of a series with an float scalar value on the left-hand side" do
  s1 = Series.from_list([1, 2, 3])

  result = Series.pow(2.0, s1)

  assert result.dtype == :float
  assert Series.to_list(result) == [2.0, 4.0, 8.0]
end
```

### PR DESCRIPTION
**Pull request by nallwhy**
This PR resolves #389.

### CODE REVIEW

**Comment by josevalim** [comment]
Looks great! What happens when you give a negative integer exponent as part of a series? Does it panic?

**Comment by nallwhy** [comment]
Yes, it panics.

```elixir
s = Explorer.Series.from_list(-1..1 |> Enum.to_list())

Explorer.Series.pow(2, s)
```

```
** (ErlangError) Erlang error: :nif_panicked
    (explorer 0.4.0-dev) Explorer.PolarsBackend.Native.s_pow_i_lhs(shape: (3,)
Series: '' [i64]
[
	-1
	0
	1
], 2)
    (explorer 0.4.0-dev) lib/explorer/polars_backend/shared.ex:17: Explorer.PolarsBackend.Shared.apply_series/3
    #cell:6q66pqxog4cjj2g2w3ojjbbg3b3qryfx:11: (file)
```

**Comment by josevalim** [comment]
Is it possible to convert the panic into a better error message? Something like “negative exponent with integer base”?

**Comment by nallwhy** [comment]
@josevalim Where should that converting be in? Elixir or Rust?

I tried that in Rust, but I don't know it is what you intended.

```rust
pub fn s_pow_i_lhs(data: ExSeries, base: u32) -> Result<ExSeries, ExplorerError> {
    let s = &data.resource.0;

    match s.strict_cast(&DataType::UInt32) {
        Ok(s) => {
            let s = s.u32()?.apply(|v| base.pow(v)).into_series();
            Ok(ExSeries::new(s))
        }
        Err(_) => {
            return Err(ExplorerError::Other(String::from(
                "negative exponent with integer base",
            )))
        }
    }
}
```

```
** (RuntimeError) Other("negative exponent with integer base")
    (explorer 0.4.0-dev) lib/explorer/polars_backend/shared.ex:21: Explorer.PolarsBackend.Shared.apply_series/3
    #cell:6q66pqxog4cjj2g2w3ojjbbg3b3qryfx:11: (file)
```

**Comment by josevalim** [comment]
I don't know either. Can you do this?

            return Err(String::from("negative exponent with integer base"))

Any ideas @philss?

**Comment by philss** [comment]
> I tried that in Rust, but I don't know it is what you intended.

This is the way we do today. We could add another error in the ExplorerError enum (at `src/error.rs`), and then treat it better at the Elixir level.
I mean: in the Elixir side, we could better format the error (remove the `Other()` part).

**Review by josevalim — APPROVED** [review]

File: native/explorer/src/series.rs [hunk 1]
```diff
@@ -659,12 +659,40 @@ pub fn s_pow(data: ExSeries, exponent: f64) -> Result<ExSeries, ExplorerError> {
...
+pub fn s_pow_i_lhs(data: ExSeries, base: u32) -> Result<ExSeries, ExplorerError> {
+    let s = &data.resource.0;
+
+    match s.strict_cast(&DataType::UInt32) {
+        Ok(s) => {
+            let s = s.u32()?.apply(|v| base.pow(v)).into_series();
+            Ok(ExSeries::new(s))
+        }
+        Err(_) => {
+            return Err(ExplorerError::Other(String::from(
```
Comment (philss) [inline native/explorer/src/series.rs:689]:
```suggestion
            Err(ExplorerError::Other(String::from(
```

**Review by philss — APPROVED** [review]

File: native/explorer/src/series.rs [hunk 2]
```diff
@@ -659,12 +659,40 @@ pub fn s_pow(data: ExSeries, exponent: f64) -> Result<ExSeries, ExplorerError> {
...
+    match s.strict_cast(&DataType::UInt32) {
+        Ok(s) => {
+            let s = s.u32()?.apply(|v| base.pow(v)).into_series();
+            Ok(ExSeries::new(s))
+        }
+        Err(_) => {
+            Err(ExplorerError::Other(String::from(
+                "negative exponent with integer base",
+            )))
+        }
```
Comment (philss) [inline native/explorer/src/series.rs:692]:
```suggestion
        Err(_) => Err(ExplorerError::Other(String::from(
            "negative exponent with integer base",
        ))),
```

**Comment by philss** [comment]
@nallwhy thank you! :purple_heart:

**Pull request merged**